### DISCUSSION TITLE
Trigger storing from provider

### DISCUSSION THREAD

**Discussion by PhilGarb** [ueberdosis/hocuspocus#627] [General]
Hi everyone!

I am wondering whether there is a way to know on the client when a document has been stored in the database. I have actions that use what is stored in the database by the Database extension. Currently when a change is made and one of these actions is executed it receives an outdated state of the document, because the document is stored after ca. 3-5 seconds of client inactivity. 
I generally really like the delay, but in these cases it leads to issues. My initial thought is that it would be perfect if the client could trigger storing the document in the database and only then performing the actual action.

**Comment by janthurau** [comment]
@PhilGarb you can probably build this using a custom database extension and our stateless message functionality. So basically after saving a document you call document.broadcastStateless, which will then send a message to the frontend and thus it knows that the data has been saved.

I just noticed that this wasn't showing on our docs because of a merging issue, I just fixed that. Here's the link: https://tiptap.dev/hocuspocus/server/hooks#on-stateless

**Reply by PhilGarb** [reply]
Hi thanks for the suggestion. That worked for me, but feels a little hacky. I wonder wether this would make for a good addition to the library. I can imagine there are many use cases where you would need to make sure the data is stored at a certain moment to run another action against it. Using the db as a source of truth vs. the websocket connection is allowing the actions to be more decoupled.
